src/Tests/DcsBios/DCSBIOSControlLocatorTests.cs

```diff
@@ -0,0 +1,363 @@
+            var dcsAircraft = DCSAircraft.GetAircraft(dcsAircraftId);
+
+            DCSBIOSControlLocator.DCSAircraft = dcsAircraft;
+            DCSBIOSControlLocator.JSONDirectory = jsonPath;
+
+            var dcsbiosOutput = DCSBIOSControlLocator.GetDCSBIOSOutput(dcsbiosControlId, dcsBiosOutputType);
+
+            Assert.Equal(dcsbiosOutput.DCSBiosOutputType, dcsBiosOutputType);
+        }
+
+        [Theory]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5, "NO_SUCH_CONTROL_ID", DCSBiosOutputType.IntegerType)]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5, "NO_SUCH_CONTROL_ID", DCSBiosOutputType.StringType)]
+        public void TestLocatorGetDCSBIOSOutputFail(string jsonPath, int dcsAircraftId, string dcsbiosControlId, DCSBiosOutputType dcsBiosOutputType)
+        {
+            jsonPath = Environment.ExpandEnvironmentVariables(jsonPath);
+            if (!Directory.Exists(jsonPath))
+            {
+                //No need to test as dcs-bios isn't installed
+                return;
+            }
+            Common.ResetEmulationModesFlag();
+
+            DCSAircraft.FillModulesListFromDcsBios(jsonPath, true);
+            var dcsAircraft = DCSAircraft.GetAircraft(dcsAircraftId);
+
+            DCSBIOSControlLocator.DCSAircraft = dcsAircraft;
+            DCSBIOSControlLocator.JSONDirectory = jsonPath;
+
+            Assert.Throws<Exception>(() => DCSBIOSControlLocator.GetDCSBIOSOutput(dcsbiosControlId, dcsBiosOutputType));
+        }
+
+        [Theory]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5, "ARC210_25KHZ_SEL", DCSBiosOutputType.IntegerType)]
+        [InlineData(@"%USERPROFILE%\Saved Games\DCS\Scripts\DCS-BIOS\doc\json", 5, "ARC210_25KHZ_SEL", DCSBiosOutputType.StringType)]
+        public void TestLocatorGetDCSBIOSOutput2(string jsonPath, int dcsAircraftId, string dcsbiosControlId, DCSBiosOutputType dcsBiosOutputType)
+        {
+            jsonPath = Environment.ExpandEnvironmentVariables(jsonPath);
+            if (!Directory.Exists(jsonPath))
+            {
+                //No need to test as dcs-bios isn't installed
+                return;
+            }
+            Common.ResetEmulationModesFlag();
+
+            DCSAircraft.FillModulesListFromDcsBios(jsonPath, true);
+            var dcsAircraft = DCSAircraft.GetAircraft(dcsAircraftId);
+
+            DCSBIOSControlLocator.DCSAircraft = dcsAircraft;
+            DCSBIOSControlLocator.JSONDirectory = jsonPath;
+
+            var dcsbiosOutput = dcsBiosOutputType switch
+            {
+                DCSBiosOutputType.IntegerType => DCSBIOSControlLocator.GetUIntDCSBIOSOutput(dcsbiosControlId),
+                DCSBiosOutputType.StringType => DCSBIOSControlLocator.GetStringDCSBIOSOutput(dcsbiosControlId),
+                _ => null
+            };
+
+            Assert.NotNull(dcsbiosOutput);
```

Comment:
Nice use of NotNull 👍   Sometimes even if I know we want to use .NotNull, I still write asserts like Assert.False(SomeVariable == null)     Shame on me :-p